th2_common/schema/factory/common_factory.py

```diff
@@ -83,23 +90,184 @@ def create_from_arguments(args=None):
+                            help='namespace in Kubernetes to find config maps related to the target')
+        parser.add_argument('--box_name',
+                            help='name of the target th2 box placed in the specified namespace in Kubernetes')
         parser.add_argument('--customConfiguration',
                             help='path to json file with custom configuration')
         result = parser.parse_args(args)
 
+        if hasattr(result, "namespace") and hasattr(result, "box_name"):
+
+            return CommonFactory.create_from_kubernetes(result.namespace, result.box_name)
+
+        else:
+
+            return CommonFactory(
+                rabbit_mq_config_filepath=CommonFactory.calculate_path(result, 'rabbitConfiguration',
+                                                                       CommonFactory.RABBIT_MQ_CONFIG_FILENAME),
+                mq_router_config_filepath=CommonFactory.calculate_path(result, 'messageRouterConfiguration',
+                                                                       CommonFactory.MQ_ROUTER_CONFIG_FILENAME),
+                grpc_router_config_filepath=CommonFactory.calculate_path(result, 'grpcRouterConfiguration',
+                                                                         CommonFactory.GRPC_ROUTER_CONFIG_FILENAME),
+                cradle_config_filepath=CommonFactory.calculate_path(result, 'cradleConfiguration',
+                                                                    CommonFactory.CRADLE_CONFIG_FILENAME),
+                custom_config_filepath=CommonFactory.calculate_path(result, 'customConfiguration',
+                                                                    CommonFactory.CUSTOM_CONFIG_FILENAME)
+            )
+
+    @staticmethod
+    def create_from_kubernetes(namespace, box_name):
+        global grpc, custom, mq, cradle, rabbit
+
+        config.load_kube_config()
+        kube_contexts = config.kube_config.list_kube_config_contexts()
+
+        v1 = client.CoreV1Api()
+
+        config_maps = v1.list_namespaced_config_map(namespace)
+        config_maps_dict = config_maps.to_dict()
+
+        service_in_namespace_service = v1.list_namespaced_service("service").to_dict()
+
+        services = v1.list_namespaced_service(namespace)
+        services_dict = services.to_dict()
+
+        config_dir = "generated_configs"
+
+        grpc_path = config_dir + "/" + CommonFactory.GRPC_ROUTER_CONFIG_FILENAME
+        custom_path = config_dir + "/" + CommonFactory.CUSTOM_CONFIG_FILENAME
+        mq_path = config_dir + "/" + CommonFactory.MQ_ROUTER_CONFIG_FILENAME
+        cradle_path = config_dir + "/" + CommonFactory.CRADLE_CONFIG_FILENAME
+        rabbit_path = config_dir + "/" + CommonFactory.RABBIT_MQ_CONFIG_FILENAME
+        dictionary_path = config_dir + "/" + "dictionary.json"
+        prometheus_path = config_dir + "/" + "prometheus.json"
+
+        for config_map in config_maps_dict['items']:
```

Comment:
Here and below, calls like *config_maps_dict ['items']* are used. It would be necessary to do some checks for the presence of the key. If it, suddenly, is not, then indicate in the log file.